src/navbar-state/navbar-state-reducer.test.tsx

```diff
@@ -0,0 +1,528 @@
+} as const;
+const testItems = [item0, item1, item2] as const;
+
+describe(navBarStateReducer.name, () => {
+  describe("items/updated action", () => {
+    it("should make honor to previously set URL", () => {
+      // Arrange
+      const state0 = {
+        currentUrl: url1,
+        isExpanded: false,
+        items: [],
+        selectedItemId: null,
+      } as const;
+      const action = {
+        type: "items/updated",
+        items: testItems,
+      } as const;
+
+      // Act
+      const state1 = navBarStateReducer(state0, action);
+
+      // Assert
+      expect(state1.items).toHaveLength(3);
+      expect(state1.items[1].isActive).toBe(true);
+
+      // When the item state does not change, the item should be the same
+      expect(state1.items[0]).toBe(item0);
+      expect(state1.items[1]).not.toBe(item1);
+      expect(state1.items[2]).toBe(item2);
+    });
+  });
+
+  describe("url/updated action", () => {
+    it("should update the active item when there is not a previously activated item", () => {
+      // Arrange
+      const state0 = {
+        currentUrl: "Unknown URL",
+        isExpanded: false,
+        items: testItems,
+        selectedItemId: null,
+      } as const;
+      const action = {
+        type: "url/updated",
+        href: url1,
+      } as const;
+
+      // Act
+      const state1 = navBarStateReducer(state0, action);
+
+      // Assert
+      expect(state1.items).toHaveLength(3);
+      expect(state1.currentUrl).toBe(url1);
+      expect(state1.isExpanded).toBe(false);
+      expect(state1.selectedItemId).toBeNull();
+      expect(state1.items).toHaveLength(3);
+      expect(state1.items[1].isActive).toBe(true);
+
+      // When the item state does not change, the item should be the same
+      expect(state1.items[0]).toBe(item0);
+      expect(state1.items[1]).not.toBe(item1);
+      expect(state1.items[2]).toBe(item2);
+    });
+
+    it("should update the active item when there is a previously activated item", () => {
+      // Arrange
+      const state0 = {
+        currentUrl: url0,
+        isExpanded: false,
+        items: [{ ...item0, isActive: true }, item1, item2],
+        selectedItemId: null,
+      } as const;
+      const action = {
+        type: "url/updated",
+        href: url1,
+      } as const;
+
+      // Act
+      const state1 = navBarStateReducer(state0, action);
+
+      // Assert
+      expect(state1.items).toHaveLength(3);
+      expect(state1.currentUrl).toBe(url1);
+      expect(state1.isExpanded).toBe(false);
+      expect(state1.selectedItemId).toBeNull();
+      expect(state1.items).toHaveLength(3);
+      expect(state1.items[1].isActive).toBe(true);
+
+      // When the item state does not change, the item should be the same
+      expect(state1.items[1]).not.toBe(item0);
+      expect(state1.items[0]).toEqual(item0);
+      expect(state1.items[1]).not.toBe(item1);
+      expect(state1.items[2]).toBe(item2);
+    });
+
+    it("should deselect item", () => {
```

Comment:
**_Cuando debe seleccionar Item?_** entiendo que seleccionar item sucede cuando el mouse deja el hacer `hover `sobre un elemento _"eso se pretende desde la UI"_ entonces; aquí para este caso la **it** no debería decir algo así como: debe deseleccionar item cuando se cambia de href?

Reply:
Creo que no es necesario aclarar eso ahí porque es parte del nombre del test:

<img width="925" alt="image" src="https://user-images.githubusercontent.com/1157864/188870674-1f9a3231-202c-4232-bac6-0ae20ba7c200.png">

_"navBarStateReducer url/updated action should deselect item"_

Creo que "cuando se cambia de href" está claramente indicado por la acción de `url/updated`.

Reply:
Ok, no estuve atento s los `describe`, gracias